packages/gatsby/src/commands/build-html.ts

```diff
@@ -53,27 +63,28 @@ const deleteRenderer = async rendererPath => {
 }
 
 const renderHTMLQueue = (
-  { workerPool, activity },
-  htmlComponentRendererPath,
-  pages
-) =>
-  new Promise((resolve, reject) => {
+  workerPool: IWorkerPool,
+  activity: IActivity,
+  htmlComponentRendererPath: string,
+  pages: string[]
+): Promise<void> =>
+  new Bluebird((resolve, reject) => {
     // We need to only pass env vars that are set programmatically in gatsby-cli
     // to child process. Other vars will be picked up from environment.
-    const envVars = Object.entries({
-      NODE_ENV: process.env.NODE_ENV,
-      gatsby_executing_command: process.env.gatsby_executing_command,
-      gatsby_log_level: process.env.gatsby_log_level,
-    })
+    const envVars = [
+      [`NODE_ENV`, process.env.NODE_ENV],
+      [`gatsby_executing_command`, process.env.gatsby_executing_command],
+      [`gatsby_log_level`, process.env.gatsby_log_level],
+    ]
 
     // const start = process.hrtime()
     const segments = chunk(pages, 50)
     // let finished = 0
 
-    Promise.map(
+    Bluebird.map(
       segments,
       pageSegment =>
-        new Promise((resolve, reject) => {
+        new Bluebird<void>((resolve, reject) => {
           workerPool
             .renderHTML({
               htmlComponentRendererPath,
```

Comment:
Could we use async/await here? (I realise this is old code)

Reply:
Yeah! Good call out. I improved a few paths that looked like async/await was better